fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Group.java

```diff
@@ -132,6 +137,26 @@ public void setBestExpressionWithStatistics(GroupExpression expression, double c
         }
     }
 
+    public Set<GroupExpression> getSatisfyRequiredGroupExpressions(PhysicalPropertySet requiredProperty) {
+        return satisfyRequiredPropertyGroupExpressions.get(requiredProperty);
+    }
+
+    public void addSatisfyRequiredPropertyGroupExpression(PhysicalPropertySet outputProperty,
+                                                          GroupExpression groupExpression) {
+        if (!satisfyRequiredPropertyGroupExpressions.containsKey(outputProperty)) {
+            satisfyRequiredPropertyGroupExpressions.put(outputProperty, Sets.newLinkedHashSet());
```

Comment:
Why need newLinkedHashSet?

Reply:
To keep order， otherwise we may get different plan even the nth is same